R/maths_correctForMemoryEffect.R

```diff
@@ -46,6 +46,30 @@ calculateMemoryCoefficients <- function(dataset) {
   # join the mean mem coeff and the mem coeff for each standard into one dataframe
   tablesToJoin <- append(memoryCoeffForEachStandard, list(meanMemoryCoefficients))
   memCoeffOutput <- reduce(tablesToJoin, full_join, by = "Inj Nr")
+
+  # remove all trailing rows from output dataframe which only contain NA values
+  memCoeffOutput <- memCoeffOutput %>%
+    filter(cumall(!is.na(memoryCoeffD18O)))
+
+  # pad mean memory coefficients with 1's if last coefficient refers to an
+  # injection number less than the maximum inj. number of the samples/standards
+  # in the other blocks
+
+  # bypass if only block 1 is passed as dataset
+  if (nrow(dataset) > nrow(filter(dataset, block == 1))) {
+
+    maxInjMeasurement <- max(
+      filter(dataset, is.na(dataset$block) | dataset$block != 1)$`Inj Nr`)
```

Comment:
This could be more intuitive
```R
maxInjMeasurement <- dataset %>%
    filter(block != 1) %>%
    select(`Inj Nr`) %>%
    max()
```
Do we even need to discount block 1 samples for this?

Reply:
This part of code is no longer needed, but let me comment that I discounted the block 1 samples here because any additional injections stemming from block 1 could only arise from the "warm-up" (=very first) standard. In such a case, no padding would be necessary because of #38.